include/swift/Sema/IDETypeChecking.h

```diff
@@ -40,6 +40,10 @@ namespace swift {
   /// \returns true on convertible, false on not.
   bool isConvertibleTo(Type T1, Type T2, DeclContext &DC);
 
+  /// \brief Determine whether a declaration is implicitly convertible to a
+  /// given type. Unlike isConvertibleTo, this handles generic types correctly.
+  bool isDeclConvertibleTo(const ValueDecl *decl, Type fromType, Type toType);
```

Comment:
Here, `fromType` is the same as `decl->getInterfaceType()`, possibly with the `self` type stripped off. I think it would be easier for the client if the signature were:

    bool isDeclConvertibleTo(const ValueDecl *decl, Type selfType, Type toType);

where `selfType` can be an instance type or metatype (when we know it and therefore might want the uncurried case), or null (leave it curried and we don't have information). That would allow isDeclConvertibleTo to perform the appropriate substitutions I mention below.